src/ScheduledTasks/Entity/Schedule.cs

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DurableTask.ScheduledTasks;
+
+/// <summary>
+/// Entity that manages the state and execution of a scheduled task.
+/// </summary>
+/// <remarks>
+/// The Schedule entity maintains the configuration and state of a scheduled task,
+/// handling operations like creation, updates, pausing/resuming, and executing the task
+/// according to the defined schedule.
+/// </remarks>
+/// <param name="logger">Logger for recording schedule operations and events.</param>
+class Schedule(ILogger<Schedule> logger) : TaskEntity<ScheduleState>
+{
+    readonly ILogger<Schedule> logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+    /// <summary>
+    /// Creates a new schedule with the specified configuration. If already exists, update it in place.
+    /// </summary>
+    /// <param name="context">The task entity context.</param>
+    /// <param name="scheduleCreationOptions">The configuration options for creating the schedule.</param>
+    /// <exception cref="ArgumentNullException">Thrown when scheduleConfigurationCreateOptions is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the schedule is already created.</exception>
+    public void CreateSchedule(TaskEntityContext context, ScheduleCreationOptions scheduleCreationOptions)
+    {
+        try
+        {
+            if (!this.CanTransitionTo(nameof(this.CreateSchedule), ScheduleStatus.Active))
+            {
+                throw new ScheduleInvalidTransitionException(scheduleCreationOptions?.ScheduleId ?? string.Empty, this.State.Status, ScheduleStatus.Active, nameof(this.CreateSchedule));
+            }
+
+            if (scheduleCreationOptions == null)
+            {
+                throw new ScheduleClientValidationException(string.Empty, "Schedule creation options cannot be null");
+            }
+
+            this.State.ScheduleConfiguration = ScheduleConfiguration.FromCreateOptions(scheduleCreationOptions);
+            this.TryStatusTransition(nameof(this.CreateSchedule), ScheduleStatus.Active);
```

Comment:
At this point `this.CanTransitionTo` has already been checked, so there is no need to check it again inside `TryStatusTransition`. 

So you could simplify this line to just `this.State.Status = ScheduleStatus.Active`